Sources/SIL/SIBParser.swift

```diff
@@ -0,0 +1,399 @@
+import Foundation
+
+let SIB_MAGIC: [UInt8] = [0xE2, 0x9C, 0xA8, 0x0E]
+
+enum SIBFileError: Error {
+  case cannotOpenFile
+  case incorrectMagic
+}
+
+enum SIBError: Error {
+  case parseError(_ reason: String)
+  case unsupported(_ what: String)
+}
+
+func loadSIBBitcode(fromPath path: String) throws -> BitcodeBlock {
+  guard let handle = FileHandle(forReadingAtPath: path) else {
+    throw SIBFileError.cannotOpenFile
+  }
+
+  var stream = Bitstream(handle.readDataToEndOfFile())
+  if (try stream.next(bytes: 4) != SIB_MAGIC) {
+    throw SIBFileError.incorrectMagic
+  }
+
+  let parser = BitcodeParser(stream)
+  return try parser.parse()
+}
+
+fileprivate func expectOne<T>(called name: String, among candidates: [T]) throws -> T {
```

Comment:
```suggestion
fileprivate func expectOne<T>(named name: String, among candidates: [T]) throws -> T {
```
It's usually "named:" rather than "called:", e.g. [`Image.init(named:)`](https://developer.apple.com/documentation/uikit/uiimage/1624146-init) :)